brainwidemap/decoding/functions/process_outputs.py

```diff
@@ -0,0 +1,382 @@
+    including subject, eid, region, test statistic, p-value, median of null distribution,
+    number of units used for decoding, fraction of weights which exceed a threshold of 0.1, 
+    and the gini index of the weights to capture sparsity.
+    Optionally returns a secondary table which also includes all the regressors, targets,
+    predictions, and weights.
+
+    Parameters
+    ----------
+    res : pandas DataFrame 
+        formatted as done in the output of 04_format_slurm.py.
+    score_name : str, optional
+        'balanced_acc_test' or 'R2_test'. The default is 'balanced_acc_test'.
+    N_PSEUDO : int, optional
+        from settings.py. The default is 200.
+    N_RUN : int, optional
+        from settings.py. The default is 10.
+    N_PSEUDO_LOWER_THRESH : scalar, optional
+        if finite, allows processing of outputs when not all N_PSEUDO nulls are present. 
+        The default is np.infty.
+    RETURN_X_Y : bool
+        Only works for scalar values per trial, e.g. not wheel-speed. 
+        returns an additional table with regressors, targets, and predictions
+        The default is False.
+    SCALAR_PER_TRIAL : bool
+        set to False if decoding multiple values per trial e.g. wheel decoding
+        The default is True.
+    SAVE_REGRESSORS : bool
+        if RETURN_X_Y then the returned xy-table's regressors columns will be empty if
+        SAVE_REGRESSORS if False.  Either way, the same checks are done on regressors',
+        representation, they are just not saved.
+        The default is True.
+
+    Returns
+    -------
+    pandas DataFrame 
+        summary of all decoding test statistics, p-values, weight analysis, and units.
+    pandas DataFrame
+        Optional, only returned if RETURN_X_Y.
+        summary of add decoder regressors, targets, and weights
+
+    '''
+    if not score_name in ['balanced_acc_test', 'R2_test']:
+        raise NotImplementedError('this score is not implemented')
+    
+    res_table = []
+    xy_table = []
+    
+    for eid in np.unique(res['eid']):
+        print(f'working on {eid}')
+        reseid = res.loc[res['eid']==eid]
+        subject = np.unique(reseid['subject'])
+        assert len(subject) == 1
+        subject = subject[0]
+        
+        #print(reseid['region'])
+        for reg in np.unique(reseid['region']):
+            # assert len(reg) == 1
+            # reg = reg[0]
+            
+            reseidreg = reseid.loc[reseid['region']==reg]
+            eidreg_probes = np.unique(reseidreg['probe'])
+            if not (len(eidreg_probes)==1):
+                print(eidreg_probes)
+                assert (eidreg_probes[0]=='probe00') and (eidreg_probes[1]=='probe01')
+                assert len(eidreg_probes)==2
+                cur_p = np.random.choice(['probe00','probe01'])
+                reseidreg = reseidreg.loc[reseidreg['probe']==cur_p]
+                
+            
+            pids = np.sort(np.unique(reseidreg['pseudo_id']))
```

Comment:
Avoid using the variable name pid as it is quite strongly associated with probe ID withing IBL

Reply:
@bensonbrandon